src/mongo/MongoTripod.class.php

```diff
@@ -657,11 +633,12 @@ protected function getDataUpdater()
     {
         if(!isset($this->dataUpdater))
         {
+            $readPreference = $this->collection->getReadPreference();
             $opts = array(
                 'defaultContext'=>$this->defaultContext,
                 OP_ASYNC=>$this->async,
                 'stat'=>$this->stat,
-                'readPreference'=>$this->collection->getReadPreference(),
+                'readPreference'=>$readPreference['type'],
```

Comment:
For the record, I think this is a good point and something we should definitely be more serious about.

I sort of wonder if it makes more sense to address in the config rather than instantiation, but I haven't through the use cases yet.